packages/sources/coinmetrics-test/src/endpoint/price-http.ts

```diff
@@ -0,0 +1,86 @@
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { ProviderResult } from '@chainlink/external-adapter-framework/util'
+import { VALID_QUOTES } from '../config'
+import { AssetMetricsEndpointTypes } from './price'
+
+export type MetricData = {
+  asset: string
+  time: string
+  ReferenceRateUSD?: string
+  ReferenceRateEUR?: string
+  ReferenceRateETH?: string
+  ReferenceRateBTC?: string
+}
+
+interface ResponseSchema {
+  data: MetricData[]
+  error?: {
+    type: string
+    message: string
+  }
+}
+
+type AssetMetricsHttpTypes = AssetMetricsEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseSchema
+  }
+}
+
+export const httpTransport = new HttpTransport<AssetMetricsHttpTypes>({
+  prepareRequests: (params, config) => {
+    return {
+      params,
+      request: {
+        baseURL: config.API_ENDPOINT,
+        url: `timeseries/asset-metrics`,
+        params: {
+          assets: [...new Set(params.map((p) => p.base.toUpperCase()))].join(','),
+          metrics: [...new Set(params.map((p) => `ReferenceRate${p.quote.toUpperCase()}`))].join(
```

Comment:
Wouldn't this build the metrics param with repeat quotes potentially? Like if params are ETH/USD and BTC/USD. Wouldn't metrics be `ReferenceRateUSD,ReferenceRateUSD` in that case?

Reply:
It shouldn't since `Set` will remove duplicates

Reply:
Oh shoot I skimmed this too quickly. Missed the Set inline